src/Microsoft.Azure.Relay/ClientWebSocketFactory.cs

```diff
@@ -10,47 +10,107 @@ namespace Microsoft.Azure.Relay
     using System.Threading;
     using System.Threading.Tasks;
 
-    interface IClientWebSocket
+    /// <summary>
+    /// Client WebSocket Interface.
+    /// </summary>
+    public interface IClientWebSocket
     {
+        /// <summary>
+        /// Client Websocket Options.
+        /// </summary>
         IClientWebSocketOptions Options { get; }
 
+        /// <summary>
+        /// Http Response Message.
+        /// </summary>
         HttpResponseMessage Response { get; }
 
+        /// <summary>
+        /// Websocket object.
+        /// </summary>
         WebSocket WebSocket { get; }
 
+        /// <summary>
+        /// Connect to a WebSocket server as an asynchronous operation.
+        /// </summary>
+        /// <param name="uri">The URI of the WebSocket server to connect to.</param>
+        /// <param name="cancellationToken">A cancellation token used to propagate notification that the operation should be canceled.</param>
+        /// <returns></returns>
         Task ConnectAsync(Uri uri, CancellationToken cancellationToken);
     }
 
-    interface IClientWebSocketOptions
+    /// <summary>
+    /// Client Websocket Options interface.
+    /// </summary>
+    public interface IClientWebSocketOptions
     {
+        /// <summary>
+        /// Gets or sets the proxy for WebSocket requests.
+        /// </summary>
         IWebProxy Proxy { get; set; }
 
+        /// <summary>
+        /// Gets or sets the WebSocket protocol keep-alive interval.
+        /// </summary>
         TimeSpan KeepAliveInterval { get; set; }
 
+        /// <summary>
+        /// Adds a sub-protocol to be negotiated during the WebSocket connection handshake.
+        /// </summary>
+        /// <param name="subProtocol">The WebSocket sub-protocol to add.</param>
         void AddSubProtocol(string subProtocol);
 
+        /// <summary>
+        /// Sets the client buffer parameters.
+        /// </summary>
+        /// <param name="receiveBufferSize">The size, in bytes, of the client receive buffer.</param>
+        /// <param name="sendBufferSize">The size, in bytes, of the client send buffer.</param>
         void SetBuffer(int receiveBufferSize, int sendBufferSize);
 
+        /// <summary>
+        /// Creates a HTTP request header and its value.
+        /// </summary>
+        /// <param name="name">The name of the HTTP header.</param>
+        /// <param name="value">The value of the HTTP header.</param>
         void SetRequestHeader(string name, string value);
     }
 
+    /// <summary>
+    /// Client WebSocket Factory interface.
+    /// </summary>
+    public interface IClientWebSocketFactory
```

Comment:
>IClientWebSocketFactory [](start = 21, length = 23)

We could consider putting KeepAliveInterval on IClientWebSocketFactory if we do the approach where there's a default HCClient.ClientWebSocketFactory (initialized to ClientWebSocketFactory.Default).

Reply:
I don't think I like this idea of mine. Probably better as you have it where HCClient and HCListener have top level KeepAliveInterval properties.

---
In reply to: [605114530](https://github.com/Azure/azure-relay-dotnet/pull/145#discussion_r605114530) [](ancestors = 605114530)